### PR TITLE
Message Archive scenario: Bot with RSC permission - Nodejs

### PR DESCRIPTION
**Pull request by NikitaJain-MSFT**
Bot with RSC permission can fetch all the messages from a group chat, archive it to text file and send it to user in personal scope.

### CODE REVIEW

File: samples/bot-groupchat-messages-withRSC/nodejs/appPackage/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# App Package
+
+Your "app package" is used to publish your app configuration to Teams by manually using the "Upload a custom app" from the app gallery in Teams.
```
Comment (v-chetsh) [inline samples/bot-groupchat-messages-withRSC/nodejs/appPackage/README.md:3]: Make sure to link your readme with root file.

File: samples/bot-groupchat-messages-withRSC/nodejs/appPackage/manifest.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      "short": "Receive groupchat messages with RSC",
+      "full": "This sample demos a bot with RSC permission"
+    },
+    "icons": {
+      "outline": "outline.png",
+      "color": "color.png"
+    },
+    "bots": [
+      {
+        "botId": "{TODO: MicrosoftAppId}",
```
Comment (v-chetsh) [inline samples/bot-groupchat-messages-withRSC/nodejs/appPackage/manifest.json:27]: Why TODO ?

File: samples/bot-groupchat-messages-withRSC/nodejs/appPackage/manifest.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    ],
+    "accentColor": "#60A18E",
+    "permissions": [ "identity", "messageTeamMembers" ],
+    "validDomains": [
+      "<<DOMAIN-NAME>>",
+      "token.botframework.com"
+    ],
+    "webApplicationInfo": {
+      "id": "{TODO: MicrosoftAppId}",
+      "resource": "api://botid-{TODO: MicrosoftAppId}",
```
Comment (v-chetsh) [inline samples/bot-groupchat-messages-withRSC/nodejs/appPackage/manifest.json:45]: Why TODO?

File: samples/bot-groupchat-messages-withRSC/nodejs/bots/dialogBot.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            await this.dialog.run(context, this.dialogState);
+
+            await next();
+        });
+    }
+
+    /**
+     * Override the ActivityHandler.run() method to save state changes after the bot logic completes.
+     */
+    async run(context) {
```
Comment (v-chetsh) [inline samples/bot-groupchat-messages-withRSC/nodejs/bots/dialogBot.js:48]: Alignment is not correct

File: samples/bot-groupchat-messages-withRSC/nodejs/dialogs/mainDialog.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    * The run method handles the incoming activity (in the form of a DialogContext) and passes it through the dialog system.
+    * If no dialog is active, it will start the default dialog.
+    * @param {*} dialogContext
+    */
+    async run(context, accessor) {
+        const dialogSet = new DialogSet(accessor);
+        dialogSet.add(this);
+        const dialogContext = await dialogSet.createContext(context);
+        const results = await dialogContext.continueDialog();
+        if (results.status === DialogTurnStatus.empty) {
```
Comment (v-chetsh) [inline samples/bot-groupchat-messages-withRSC/nodejs/dialogs/mainDialog.js:45]: One line space before if

File: samples/bot-groupchat-messages-withRSC/nodejs/dialogs/mainDialog.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    sendFileCard(filename, filesize) {
+        const consentContext = { filename: filename };
+
+        const fileCard = {
+            description: 'This is the file I want to send you',
+            sizeInBytes: filesize,
+            acceptContext: consentContext,
+            declineContext: consentContext
+        };
+        const asAttachment = {
```
Comment (v-chetsh) [inline samples/bot-groupchat-messages-withRSC/nodejs/dialogs/mainDialog.js:119]: One line space before it

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Reviewed your PR and provided few comments.
Please look into it.

File: samples/bot-groupchat-messages-withRSC/nodejs/appPackage/README.md (hunk 1 above)
Comment (NikitaJain-MSFT) [reply samples/bot-groupchat-messages-withRSC/nodejs/appPackage/README.md:3]: Readme is linked with root readme file

**Review by v-chetsh — APPROVED** [review]
PR looks good, approving it.

File: README.md [hunk 7]
```diff
@@ -218,6 +218,7 @@ Here are all the samples which are deployed on Microsoft Azure and you can try i
 [messageswithrsc#csharp]:samples/bot-receive-channel-messages-withRSC/csharp
 [messageswithrsc#js]:samples/bot-receive-channel-messages-withRSC/nodejs
 [app-in-meeting#cs]:samples/app-in-meeting/csharp
+[fetchgroupchatmessages#nodejs]:samples/bot-groupchat-messages/nodejs
```
Comment (Wajeed-msft) [inline README.md:221]: Could you please rename this to bot-archive-groupchat-messages?

**Review by Wajeed-msft — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -64,7 +64,7 @@ Here are all the samples which are deployed on Microsoft Azure and you can try i
 |9| Universal bots  | Teams catering bot demonstrating how to use Universal bots in Teams | [View](samples/bot-teams-catering/csharp) |  | |
 |10| Sequential workflow adaptive cards | Demonstrating on how to implement sequential flow, user specific view and upto date adaptive cards in bot. | [View][sequential#workflow#csharp] | [View][sequential#workflow#js]
 |11| Channel messages with RSC permissions | Demonstrating on how a bot can receive all channel messages with RSC without @mention. | [View][messageswithrsc#csharp] |[View][messageswithrsc#js]
-
+|12| Archive groupchat messages | Demonstrating on how a bot can archive groupchat messages and send it to user as a file. |  |[View][fetchgroupchatmessages#nodejs]|
```
Comment (Wajeed-msft) [inline README.md:67]: Let's move this under Scenario specific samples: https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/v-nikija/bot-groupchat-rsc-nodejs#scenario-specific-samples

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**